runtime/onert/backend/trix/DevContext.cc

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DevContext.h"
+
+#include "Convert.h"
+
+namespace onert
+{
+namespace backend
+{
+namespace trix
+{
+
+DevContext::DevContext() : _dev_handles{}, _model_ids{}, _meta_map{}
+{
+  auto dev_count = getnumNPUdeviceByType(NPUCOND_TRIV2_CONN_SOCIP);
+  if (dev_count <= 0)
+  {
+    throw std::runtime_error("Unable to find TRIX NPU device");
+  }
+
+  // Get NPU device handles
+  for (int i = 0; i < dev_count; ++i)
+  {
+    npudev_h handle;
+    if (getNPUdeviceByType(&handle, NPUCOND_TRIV2_CONN_SOCIP, i) < 0)
+    {
+      throw std::runtime_error("Failed to get TRIX NPU device handle");
+    }
+    _dev_handles.emplace_back(handle);
+  }
+
+  // NOTE Do not change the number of threads as long as jobs in thread call
+  //      the synchronous APIs such as submitNPU_request()
+  _batch_thread_pool = std::make_unique<BatchThreadPool>(_dev_handles.size());
```

Comment:
In case of multiple models, there may be a problem having multiple `BatchThreadPool` in current architecture. It would be good to apply the singleton pattern to prevent this problem, but it depends on how the daemon is implemented.

Reply:
I will add a comment about it as next PR.